source/Table/Table.js

```diff
@@ -403,17 +403,17 @@ export default class Table extends PureComponent {
         ? SortDirection.ASC
         : SortDirection.DESC
 
-      const onClick = () => {
+      const onClick = (evt) => {
         sortEnabled && sort({
           sortBy: dataKey,
           sortDirection: newSortDirection
         })
-        onHeaderClick && onHeaderClick({ columnData, dataKey })
+        onHeaderClick && onHeaderClick({ evt, columnData, dataKey })
       }
 
       const onKeyDown = (event) => {
         if (event.key === 'Enter' || event.key === ' ') {
-          onClick()
+          onClick(event)
```

Comment:
This is a bit weird- though I understand why you did it. I guess it's okay since people can ignore the event or check its `type`